### ISSUE TITLE
Another Column Limit Issue

### ISSUE THREAD

**Issue by yarrick19** [rstudio/sparklyr#264]
I'm excited about sparklyr but unfortunately keep running into various issues.  One of these appears to be issues that occur when working with data frames with a large number of columns.  

Below is an example that creates a dataset with 360 columns.  This gets partitioned into a test and train set using the sdf_partition function.  Any subsequent action on the resulting tables (train_tbl and test_tbl) result in an error (detail shown below).  If I run the exact same code with less than 360 columns, the code completes without error.

This is being run on an AWS EMR cluster via the preview edition of RStudio.

Code:

```
library(sparklyr)
library(dplyr)



# Connect to Spark
Sys.setenv(SPARK_HOME = "/usr/lib/spark")
sc <- spark_connect(master = "yarn-client", version = '2.0.0')


# Simulate Dataset

df <- as.data.frame(matrix(0:1, ncol = 360, nrow = 400))
sdf <- copy_to(sc, df, "df_test", overwrite = TRUE, memory = FALSE)


# Split data into train and test groups

train_test_tbl <- sdf %>%
  sdf_partition(train = 0.8, test = 0.2, seed = 100)

train_tbl <- train_test_tbl$train
test_tbl <- train_test_tbl$test


# Look at counts grouped by the first variable

sdf %>% group_by(V1) %>% summarise(ct = n())

train_tbl %>% group_by(V1) %>% summarise(ct = n())
test_tbl %>% group_by(V1) %>% summarise(ct = n())
```

Error Message:

```
Error: org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 272.0 failed 4 times, most recent failure: Lost task 0.3 in stage 272.0 (TID 6793, ip-172-31-7-73.us-west-2.compute.internal): java.util.concurrent.ExecutionException: java.lang.Exception: failed to compile: org.codehaus.janino.JaninoRuntimeException: Code of method "(Lorg/apache/spark/sql/catalyst/InternalRow;Lorg/apache/spark/sql/catalyst/InternalRow;)I" of class "org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificOrdering" grows beyond 64 KB
/* 001 */ public SpecificOrdering generate(Object[] references) {
/* 002 */   return new SpecificOrdering(references);
/* 003 */ }
/* 004 */
/* 005 */ class SpecificOrdering extends org.apache.spark.sql.catalyst.expressions.codegen.BaseOrdering {
/* 006 */
/* 007 */   private Object[] references;
/* 008 */
/* 009 */
/* 010 */
/* 011 */   public SpecificOrdering(Object[] references) {
/* 012 */     this.references = references;
/* 013 */
/* 01
```

Session Info:

```
> sessionInfo()
R version 3.2.2 (2015-08-14)
Platform: x86_64-redhat-linux-gnu (64-bit)
Running under: Amazon Linux AMI 2016.03

locale:
 [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C               LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
 [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8    LC_PAPER=en_US.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C             LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] dplyr_0.5.0     sparklyr_0.4.15

loaded via a namespace (and not attached):
 [1] readr_1.0.0     lazyeval_0.2.0  magrittr_1.5    R6_2.2.0        assertthat_0.1  rprojroot_1.0-2 parallel_3.2.2 
 [8] DBI_0.5-1       tools_3.2.2     withr_1.0.2     yaml_2.1.13     tibble_1.2      Rcpp_0.12.7     config_0.2     
[15] digest_0.6.10 
```

**Comment by kevinushey** [comment]
Thanks for sharing a reproducible example! I can reproduce this on a local Spark cluster as well.

It looks like this is an issue in Spark itself -- see here: https://issues.apache.org/jira/browse/SPARK-16845. Unfortunately, the tables generated by Spark's `randomSplit` function here produce this error (which `sparklyr` uses under the hood).

**Comment by kevinushey** [comment]
One workaround, if appropriate, would be to transform your data from wide to long (assuming you have many columns that can be collected in such a way). We're planning to add some more tools to `sparklyr` to make reshaping of Spark Datasets a bit easier.

**Comment by javierluraschi** [comment]
Seems this was fixed in `Spark 2.2`, haven't tried `Spark 2.1` but it might work there as well.

```
> train_tbl %>% group_by(V1) %>% summarise(ct = n())
# Source:   lazy query [?? x 2]
# Database: spark_connection
     V1    ct
  <int> <dbl>
1     1   163
2     0   161
> test_tbl %>% group_by(V1) %>% summarise(ct = n())
# Source:   lazy query [?? x 2]
# Database: spark_connection
     V1    ct
  <int> <dbl>
1     1    37
2     0    39
```
